Content/ContentTypeResolver/PageSelectionResolver.php

```diff
@@ -55,6 +56,9 @@ public function resolve($data, PropertyInterface $property, string $locale, arra
             $property->getParams()
         );
 
+        // the PageSelectionContainer resolves the data defined in the $params using the default content types
+        // for example, this means that the result contains an array of media entities instead of an array of ids
+        // TODO: find a solution that returns the unresolved data which can be passed to the resolvers of this bundle
```

Comment:
I think the issue is in the PageSerializer and is this not this todo then: https://github.com/sulu/SuluHeadlessBundle/blob/36aa5118b093cf9af9746aa8a7ba990239445e81/Content/Serializer/PageSerializer.php#L53

Reply:
I think there are two issues unfortunately 🙈 

1. The `PageSelectionContainer` does not return a structure, but returns the already resolved data. This data is resolved with the default sulu content types, not the resolvers of this bundle. We should probably not use the `PageSelectionContainer` here and just load the structure

2. The `PageSerializer` should not reimplement how properties of a structure are resolved, but should use the `StructureResolver` to do that.

Reply:
Okay I understand think the issues is from my side that the `PageSelectionContainer` is used. We should avoid any connection to this kind of services. That it returns the resolved data I see not an issue if it would return it correctly :)